services/file_append.proto

```diff
@@ -19,31 +27,60 @@ package proto;
  * limitations under the License.
- * ‍
  */
 
 option java_package = "com.hederahashgraph.api.proto.java";
 // <<<pbj.java_package = "com.hedera.hapi.node.file">>> This comment is special code for setting PBJ Compiler java package
 option java_multiple_files = true;
 
-
 import "basic_types.proto";
 
 /**
- * Append the given contents to the end of the specified file. If a file is too big to create with a
- * single FileCreateTransaction, then it can be created with the first part of its contents, and
- * then appended as many times as necessary to create the entire file. This transaction must be
- * signed by all initial M-of-M KeyList keys. If keys contains additional KeyList or ThresholdKey
- * then M-of-M secondary KeyList or ThresholdKey signing requirements must be meet. 
+ * A transaction body for an `appendContent` transaction.<br/>
+ * This transaction body provides a mechanism to append content to a "file" in
+ * network state. Hedera transactions are limited in size, but there are many
+ * uses for in-state byte arrays (e.g. smart contract bytecode) which require
+ * more than may fit within a single transaction. The `appendFile` transaction
+ * exists to support these requirements. The typical pattern is to create a
+ * file, append more data until the full content is stored, verify the file is
+ * correct, then update the file entry with any final metadata changes (e.g.
+ * adding threshold keys and removing the initial upload key).
+ *
+ * Each append transaction MUST remain within the total transaction size limit
+ * for the network (typically 6144 bytes).<br/>
+ * The total size of a file MUST remain within the maximum file size limit for
+ * the network (typically 1048576 bytes).
+ *
+ * #### Signature Requirements
+ * Append transactions MUST have signatures from _all_ keys in the `KeyList`
+ * assigned to the `keys` field of the file.<br/>
+ * See the [File Service](#FileService) specification for a detailed
+ * explanation of the signature requirements for all file transactions.
+ *
+ * ### Block Stream Effects
+ * None
  */
 message FileAppendTransactionBody {
     /**
-     * The file to which the bytes will be appended
+     * A file identifier.<br/>
+     * This identifies the file to which the `contents` will be appended.
+     * <p>
+     * This field is REQUIRED.<br/>
+     * The identified file MUST exist.<br/>
+     * The identified file MUST NOT be larger than the current maximum file
+     * size limit.<br/>
+     * The identified file MUST NOT be deleted.<br/>
+     * The identified file MUST NOT be immutable.
      */
     FileID fileID = 2;
 
     /**
-     * The bytes that will be appended to the end of the specified file
+     * An array of bytes to append.<br/>
+     * <p>
+     * This content SHALL be appended to the identified file if this
+     * transaction succeeds.<br/>
+     * This field is REQUIRED.<br/>
+     * This field MUST NOT be empty.
```

Comment:
I'm not seeing any check in the code that this field is not empty. I don't know why you would want to append empty content, but AFAICT it would work. @povolev15 Could a fileAppend with empty content succeed?

Reply:
This was taken from the existing documentation for the field; is the documentation incorrect, or are we missing a validation?